File: ObjectRequestBrokerCS/ORB/vson/FieldUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;

namespace ORB.vson
{
    public static class FieldUtils
    {

        private static readonly ISet<Type> WRAPPER_PRIMITIVES = new HashSet<Type> { typeof(Boolean), typeof(Char), typeof(Byte), typeof(Int16), typeof(Int32), typeof(Int64), typeof(Single), typeof(Double), typeof(void) };


        public static bool IsPrimitive(Type mClass)
        {
            return WRAPPER_PRIMITIVES.Contains(mClass) || mClass.IsPrimitive;
        }

        private static readonly IDictionary<Type, Type> builtInMap = new Dictionary<Type, Type>();

        static FieldUtils()
        {
            builtInMap[typeof(Int32)] = typeof(int);
            builtInMap[typeof(Int64)] = typeof(long);
            builtInMap[typeof(Double)] = typeof(double);
            builtInMap[typeof(Single)] = typeof(float);
            builtInMap[typeof(Boolean)] = typeof(bool);
            builtInMap[typeof(Char)] = typeof(char);
            builtInMap[typeof(Byte)] = typeof(byte);
            builtInMap[typeof(void)] = typeof(void);
            builtInMap[typeof(Int16)] = typeof(short);
        }

        public static Type AttemptPrimitiveConvesion(Type mClass)
        {
            Type retVal;
            return !builtInMap.TryGetValue(mClass, out retVal) ? mClass : retVal;
        }


        /// <summary>
        /// Returns all fields of class {@code mClass} including the fields of all its subclasses.
        /// </summary>
        /// <param name="mClass"> the class whose fields (including its subclass') will be returned. </param>
        /// <returns> an array containing the fields of {@code mClass} and the fields of all its subclasses. </returns>
        public static FieldInfo[] GetAllFields(Type mClass)
        {
            FieldInfo[] mClassFields = new FieldInfo[] { };
            while (mClass != null && mClass != typeof(object))
            {
                mClassFields = mClassFields.Concat(mClass.GetFields(BindingFlags.DeclaredOnly | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance)).ToArray();
                mClass = mClass.BaseType;
            }
            return mClassFields;
        }

    }

}

File: ObjectRequestBrokerJava/src/main/java/ORB/vson/FieldUtils.java
package ORB.vson;

import java.lang.reflect.Field;
import java.util.*;
import java.util.stream.Stream;

public class FieldUtils {

    private static final Set<Class> WRAPPER_PRIMITIVES = new HashSet<>(Arrays.asList(
            Boolean.class, Character.class, Byte.class, Short.class, Integer.class, Long.class, Float.class, Double.class, Void.class));


    public static boolean isPrimitive(Class mClass) {
        return WRAPPER_PRIMITIVES.contains(mClass) || mClass.isPrimitive();
    }

    private static Map<Class,Class> builtInMap = new HashMap<>();

    static {
        builtInMap.put(Integer.class, Integer.TYPE );
        builtInMap.put(Long.class, Long.TYPE );
        builtInMap.put(Double.class, Double.TYPE );
        builtInMap.put(Float.class, Float.TYPE );
        builtInMap.put(Boolean.class, Boolean.TYPE );
        builtInMap.put(Character.class, Character.TYPE );
        builtInMap.put(Byte.class, Byte.TYPE );
        builtInMap.put(Void.class, Void.TYPE );
        builtInMap.put(Short.class, Short.TYPE );
    }

    public static Class attemptPrimitiveConvesion(Class mClass){
        return builtInMap.getOrDefault(mClass, mClass);
    }


    /** Returns all fields of class {@code mClass} including the fields of all its subclasses.
     *
     * @param mClass the class whose fields (including its subclass') will be returned.
     * @return an array containing the fields of {@code mClass} and the fields of all its subclasses.
     */
    public static Field[] getAllFields(Class<?> mClass) {
        Field[] mClassFields = new Field[]{};
        while ( mClass != null && mClass != Object.class){
            mClassFields = Stream.concat(Arrays.stream(mClassFields), Arrays.stream(mClass.getDeclaredFields()))
                    .toArray(Field[]::new);
            mClass = mClass.getSuperclass();
        }
        return mClassFields;
    }

}
